app/src/main/java/com/teamounce/ounce/RetrofitService.kt

```diff
@@ -0,0 +1,19 @@
+package com.teamounce.ounce
+
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object RetrofitService {
+    private const val BASE_URL = "http://15.165.252.145:8080"
+
+    private fun getInstance(): Retrofit{
+        return Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+    }
+
+    fun <T> create(service : Class<T>) : T {
+        return getInstance().create(service)
+    }
```

Comment:
지난번에 말한거 내가 밑에다가 어떻게 수정할지 달아놨으니까 참고해서 넣어줘
```suggestion
    private var serviceInstance: RetrofitServiceClass? = null
    fun getRetrofitService(): RetrofitServiceClass = serviceInstance ?: synchronized(this) {
        serviceInstance ?: getInstacne().create(RetrofitServiceClass::class.java).apply {
            serviceInstance = this
        }
    }
```

여기서 중요한 점! 
- 이렇게 전체 프로젝트에서 단 하나의 객체를 만들어나가는 패턴을 Singleton 패턴이라 한다
- synchronized(this)를 하면 다양한 스레드에서 이 객체를 동시에 접근하려고 할 때
  ???: 어 너네 순서 잘 지키고 들어와 하면서 순서 정렬 시켜주기 때문에 **스레드 안정성**을 높여준다
- 하나의 객체만을 만들고 이 객체만을 이용해서 서버통신을 하기 때문에 객체 생성에 필요한 시스템 자원들을 "상당히" 아껴준다. 이 방법을 응용해서 서버통신 객체를 어떻게 만들어야 할 지 생각해봐
- 그리고 서버통신을 객체를 만들 때 하나의 인터페이스에 모든 코드를 갔다놓는게 좋을까 아니면 여러 파일로 분산시켜서 담는게 좋을까? 너의 생각을 들어보고 싶어

Reply:
- 스레드 안정성 메모메모메-모....
- 이번에 서버통신 객체 만들 때, 하나의 인터페이스에 다 모아놓고 했는데 .. 음 .... 이게 아닌가?!